src/indexer.cc

```diff
@@ -995,6 +995,24 @@ void indexDeclaration(CXClientData client_data, const CXIdxDeclInfo* decl) {
   if (!db)
     return;
 
+
+  // update the file language
+  // TODO: only do this when |is_first_ownership| in ConsumeFile is true
+  switch (clang_getCursorLanguage(decl->cursor)) {
+  case CXLanguage_C:
+    db->language = "c";
+    break;
+  case CXLanguage_CPlusPlus:
+    db->language = "c++";
+    break;
+  case CXLanguage_ObjC:
+    db->language = "objc";
+    break;
+  case CXLanguage_Invalid:
+    db->language = "invalid";
+    break;
+  }
```

Comment:
Maybe we should store an enum value instead, and do a comparison here, to "upgrade" a previous definition of this files language to `c++` from `c`, since it seems that clang pegs C compliant declarations as C. With a comparison, we could make sure to keep the "greatest" language in each file instead of the last one